File: src/main/java/tech/zolhungaj/amqapi/servercommands/social/FriendSocialStatusUpdate.java
package tech.zolhungaj.amqapi.servercommands.social;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;
import tech.zolhungaj.amqapi.servercommands.objects.PlayerGameState;
import tech.zolhungaj.amqapi.servercommands.objects.PlayerStatus;

import java.util.Optional;

@CommandType("friend social status change")
public record FriendSocialStatusUpdate(
        String name,
        @Json(name = "socialStatus") PlayerStatus status,
        Optional<PlayerGameState> gameState
){}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/social/FriendSocialStatusUpdate.kt
package tech.zolhungaj.amqapi.servercommands.social

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.PlayerGameState
import tech.zolhungaj.amqapi.servercommands.objects.PlayerStatus

@JvmRecord
@CommandType("friend social status change")
data class FriendSocialStatusUpdate(
    @Json(name = "name")
    val playerName: String,
    @Json(name = "socialStatus")
    val status: PlayerStatus = PlayerStatus.OFFLINE,
    val gameState: PlayerGameState?
)
